src/Components/SceneList/Internal/SceneDialog.tsx

```diff
@@ -195,35 +203,49 @@ const SceneDialog: React.FC<ISceneDialogProps> = ({
      * when it is mounted again when we switch tabs back to it again
      */
     useEffect(() => {
-        if (selected3DFilePivotItem === SelectionModeOf3DFile.FromContainer) {
+        if (
+            state.selected3DFilePivotItem ===
+            SelectionModeOf3DFile.FromContainer
+        ) {
             put3DFileBlob.cancelAdapter();
-            setIsSelectedFileExistInBlob(false);
-            setIsOverwriteFile(false);
-            setSelectedFile(null);
+            dispatch({
+                type: RESET_FILE,
+                payload: {
+                    isSelectedFileExistInBlob: false,
+                    isOverwriteFile: false,
+                    selectedFile: null
+                }
+            });
         }
-    }, [selected3DFilePivotItem]);
+    }, [state.selected3DFilePivotItem]);
 
     const handleNameChange = useCallback(
-        (e) => {
+        (e, newValue?: string) => {
             if (sceneToEdit) {
                 const selectedSceneCopy: IScene = deepCopy(sceneRef.current);
                 selectedSceneCopy.displayName = e.currentTarget.value;
```

Comment:
let's not use any `e.currentTarget` in this file. You can get the values off the second prop.